src/components/PBSCPasswordStrength.js

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+const strengthListDictionary = [
+  {
+    strength: 'EMPTY',
+    color: ['#C4C4C4', '#C4C4C4', '#C4C4C4', '#C4C4C4', '#C4C4C4'],
+    strengthLevel: 0,
+    suggestions: [],
+  },
+  {
+    strength: 'WEAK',
+    color: ['#DC2619', '#C4C4C4', '#C4C4C4', '#C4C4C4', '#C4C4C4'],
+    strengthLevel: 1,
+    suggestions: [],
+  },
+  {
+    strength: 'FAIR',
+    color: ['#DC2619', '#FFA114', '#C4C4C4', '#C4C4C4', '#C4C4C4'],
+    strengthLevel: 2,
+    suggestions: [],
+  },
+  {
+    strength: 'GOOD',
+    color: ['#DC2619', '#FFA114', '#76BC21', '#C4C4C4', '#C4C4C4'],
+    strengthLevel: 3,
+    suggestions: [],
+  },
+  {
+    strength: 'STRONG',
+    color: ['#DC2619', '#FFA114', '#76BC21', '#349724', '#C4C4C4'],
+    strengthLevel: 4,
+    suggestions: [],
+  },
+  {
+    strength: 'VERY_STRONG',
+    color: ['#DC2619', '#FFA114', '#76BC21', '#349724', '#287C1A'],
+    strengthLevel: 5,
+    suggestions: [],
+  },
+];
+
+const PBSCPasswordStrength = (props) => {
+  const {
+    width = '80%',
+    style,
+    isValid,
+    value,
+    passwordStrengthValidation,
+  } = props;
+
+  const [passwordStrength, setPasswordStrength] = useState(
+    strengthListDictionary[0]
+  );
+
+  useEffect(() => {
+    if (value !== '') {
+      const findStrength = strengthListDictionary.find(
+        (list) => list.strength === passwordStrengthValidation.strength
+      );
+      if (findStrength) {
+        setPasswordStrength({
+          ...findStrength,
+          suggestions: passwordStrengthValidation.suggestions,
+        });
+        isValid(passwordStrengthValidation.valid);
+      }
+    } else {
+      setPasswordStrength(strengthListDictionary[0]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]);
+
+  const getSuggestionsText = () => {
+    let suggestionsText = '';
+    passwordStrength.suggestions.map((suggestion) => {
+      if (suggestionsText === '') {
+        suggestionsText = suggestion;
+      } else {
+        suggestionsText = `${suggestionsText} ${suggestion}`;
+      }
+    });
+    return suggestionsText;
+  };
+
+  return (
+    <View style={[{ width }, style]}>
+      <View style={styles.titleWrapper}>
+        <Text>{`Password strength `}</Text>
```

Comment:
hmm i think we should translate this

Reply:
I made it a prop, english by default. The react-native-ui should not offer translation.